server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -250,11 +242,8 @@ public void updateCachedBlockIds(String appId, int shuffleId, ShufflePartitioned
   }
 
   public Roaring64NavigableMap getCachedBlockIds(String appId, int shuffleId) {
-    Map<Integer, Roaring64NavigableMap> shuffleIdToBlockIds = cachedBlockIds.get(appId);
-    if (shuffleIdToBlockIds == null) {
-      LOG.warn("Unexpected value when getCachedBlockIds for appId[" + appId + "]");
-      return Roaring64NavigableMap.bitmapOf();
-    }
+    Map<Integer, Roaring64NavigableMap> shuffleIdToBlockIds = shuffleTaskInfo
```

Comment:
Log should be kept to trace unexpected situation.

Reply:
`shuffleIdToBlockIds` was initialized in `shuffleTaskInfo`, it is not a null value and only possible to have an empty map. I think there should be no logs here. WDYT?

Reply:
Reduce task maybe request cachedBlockIds after task info was removed because of expired.
In such case, null will be returned, and it's better to add log to trace it for problem analysis.

Reply:
```
if (!shuffleToCachedBlockIds.isEmpty()) {
  storageManager.removeResources(appId, shuffleToCachedBlockIds.keySet());
}
shuffleTaskInfo.remove(appId);
```
In here, it will remove task info because of expired. If `shuffleIdToBlockIds` don't have this app, it will create a new `ShuffleTaskInfo`, and `getCachedBlockIds`  will return a `ConcurrentHashMap`. Null should not be returned.